### PR TITLE
Mod page

### LINKED ISSUE #881: /mod page
- Details about the mod
- Videos with the mod
- Link to github
- Rat checker

### PR DESCRIPTION
**Pull request by matthias-luger**
fixes #881

### CODE REVIEW

**Comment by matthias-luger** [comment]
@Ekwav The page is finished (UI-Side) but these things are still open:
 - RAT checking API returns CORS error. => Could/should be fixed in production?
 - More up-to-date youtube video / tutorial => Do you plan on making one, or should we still use Tentamens'

**Comment by Ekwav** [comment]
> @Ekwav The page is finished (UI-Side) but these things are still open:
> 
> * RAT checking API returns CORS error. => Could/should be fixed in production?
> * More up-to-date youtube video / tutorial => Do you plan on making one, or should we still use Tentamens'

* Talked to vytal, the api should work correctly now
* The installation from Tentamens is still up to date and will not change unless forge updates their website. Additionally you can link https://www.youtube.com/watch?v=dv66Y_cra8U&list=PLDpPmxIcq9tAssQlyJMBlSmSg5JOpq699

File: components/RatChecker/RatChecker.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                <p style={{ color: 'white', fontSize: 'larger' }}>
+                    This mod file is not known. For further information check{' '}
+                    <a target="_blank" rel="noreferrer" href="https://isthisarat.com/">
+                        https://isthisarat.com/
+                    </a>
+                </p>
+            )
+        }
+        if (checkingResult.rat.includes('Yes')) {
+            return <p style={{ color: 'red', fontSize: 'larger' }}>This mod file seems to be a rat. Be careful!</p>
```
Comment (Ekwav) [inline components/RatChecker/RatChecker.tsx]:
```suggestion
            return <p style={{ color: 'red', fontSize: 'larger' }}>This mod is a known rat. We recommend against using it!</p>
```

File: components/RatChecker/RatChecker.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        }
+        if (checkingResult.rat.includes('No')) {
+            return <p style={{ color: 'lime', fontSize: 'larger' }}>This mod file doesn't seem to be a rat.</p>
```
Comment (Ekwav) [inline components/RatChecker/RatChecker.tsx]:
```suggestion
            return <p style={{ color: 'lime', fontSize: 'larger' }}>No harmful code was found in this mod. It should be safe to use.</p>
```

File: components/ModDetails/ModDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+function ModDetails() {
+    return (
+        <>
+            <Card>
+                <Card.Header>
+                    <Card.Title>Details</Card.Title>
+                </Card.Header>
+                <Card.Body>
+                    <p>We provide a mod as a quality of life feature to use the flipper in minecraft directly.</p>
+                    <p>To use the mod you will need to link it to the website and verify your Minecraft account.</p>
```
Comment (Ekwav) [inline components/ModDetails/ModDetails.tsx]: The mod does way more than just provide flips in game. Also you do NOT need to link and/or verify.
```suggestion
                    <p>Say goodbye to window switching. The cofl mod provides you with data directly in Hypixel Skyblock. You can access price paid, the ah flipper, median, lbin, bazaar prices and more.</p>
                    <h3>Installation TLDR
                    <ol>
                    <li>Get and install Forge 1.8.9 (if you don't have already watch the video below)</li>
                    <li>Download and put the mod into your mods folder</li>
                    <li>Launch Minecraft and join Skyblock</li>
                    <li>Get a list of commands with /cofl help or watch the <a href="https://www.youtube.com/watch?v=dv66Y_cra8U&list=PLDpPmxIcq9tAssQlyJMBlSmSg5JOpq699">YouTube playlist</a> explaining them</li>
                    </ol>
```

File: components/ModDetails/ModDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                            Discord
+                        </a>{' '}
+                        in the #mod-releases channel.
+                    </p>
+                    <p>
+                        The entire mod is open source and you can check/compile it yourself here:{' '}
+                        <a target="_blank" rel="noreferrer" href="https://github.com/Coflnet/SkyblockMod">
+                            https://github.com/Coflnet/SkyblockMod
+                        </a>
+                    </p>
```
Comment (Ekwav) [inline components/ModDetails/ModDetails.tsx]:
```suggestion
                    <p>
                         Mod downloads including source code can be found on the <a target="_blank" rel="noreferrer" href="https://github.com/coflnet/skyblockmod/releases  ">
                            Cofl Mod Github release page
                        </a>
                    </p>
```

File: components/RatChecker/RatChecker.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                    <Card.Title>Rat Checker</Card.Title>
+                </Card.Header>
+                <Card.Body>
+                    <p>
+                        Please only download the mod from our{' '}
+                        <a target="_blank" rel="noreferrer" href="https://discord.gg/wvKXfTgCfb">
+                            Discord
+                        </a>
+                        . The download link is in the #mod-releases channel.
+                    </p>
```
Comment (Ekwav) [inline components/RatChecker/RatChecker.tsx]:
```suggestion
```
This is for general rat checking, a link to our mod should only be a sideeffect not the first line.  
Also only download from the github https://github.com/coflnet/skyblockmod/releases

File: components/RatChecker/RatChecker.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                    {!isChecking ? (
+                        <div>
+                            <p>If you still want to check if a mod file is a RAT (remote access trojan), you can upload the file here:</p>
```
Comment (Ekwav) [inline components/RatChecker/RatChecker.tsx]:
```suggestion
                            <p>Received a suspicious (or any other mod) that you want to check? Select it here and get quick results for known rats or legitimate mods (you can select multiple at once):</p>
```

File: components/RatChecker/RatChecker.tsx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                        getLoadingElement(<p>Checking file</p>)
+                    )}
+                    <hr />
+                    {checkingResult && !isChecking ? (
+                        <div>
+                            <h5>Result:</h5>
+                            {getResultElement()}
+                            <p style={{ float: 'right' }}>
+                                <small>
+                                    Files are checked by{' '}
```
Comment (Ekwav) [inline components/RatChecker/RatChecker.tsx]:
```suggestion
                                    File check is provided by{' '}
```

File: components/RatChecker/RatChecker.tsx [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                        </a>
+                        . The download link is in the #mod-releases channel.
+                    </p>
+                    {!isChecking ? (
+                        <div>
+                            <p>If you still want to check if a mod file is a RAT (remote access trojan), you can upload the file here:</p>
+                            <Form.Control type="file" className={'form-control'} ref={ratFileInput} onChange={onFileUpload} />
+                        </div>
+                    ) : (
+                        getLoadingElement(<p>Checking file</p>)
```
Comment (Ekwav) [inline components/RatChecker/RatChecker.tsx:120]: please allow checking multiple files at once

File: components/ModDetails/ModDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+                            allowFullScreen
+                        ></iframe>
+                    </div>
+                    <p style={{ marginTop: '50px' }}>
+                        We also have a playlist on how to use different ingame commands:
+                        <br />
+                        <a target="_blank" rel="noreferrer" href="https://www.youtube.com/watch?v=dv66Y_cra8U&list=PLDpPmxIcq9tAssQlyJMBlSmSg5JOpq699">
+                            {'https://www.youtube.com/watch?v=dv66Y_cra8U&list=PLDpPmxIcq9tAssQlyJMBlSmSg5JOpq699'}
+                        </a>
+                    </p>
```
Comment (Ekwav) [inline components/ModDetails/ModDetails.tsx]:
```suggestion
```

File: components/ModDetails/ModDetails.tsx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+                    <div style={{ display: 'flex', justifyContent: 'center' }}>
+                        <iframe
+                            width="560"
+                            height="315"
+                            src="https://www.youtube.com/embed/qwnvobFTcS0"
+                            title="YouTube video player"
+                            frameBorder="0"
+                            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+                            allowFullScreen
+                        ></iframe>
```
Comment (Ekwav) [inline components/ModDetails/ModDetails.tsx:43]: can this be made bigger when clicked? Like scaling it to full width of the container

File: components/ModDetails/ModDetails.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import { Card } from 'react-bootstrap'
+
+function ModDetails() {
+    return (
+        <>
+            <Card>
+                <Card.Header>
+                    <Card.Title>Details</Card.Title>
```
Comment (Ekwav) [inline components/ModDetails/ModDetails.tsx]:
```suggestion
                    <Card.Title>Cofl skyblock mod details</Card.Title>
```

File: pages/mod.tsx [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+import { getHeadElement } from '../utils/SSRUtils'
+
+export default function Mod() {
+    return (
+        <div className="page">
+            {getHeadElement('Mod')}
+            <Container>
+                <h2>
+                    <NavBar />
+                    Mod
```
Comment (Ekwav) [inline pages/mod.tsx]:
```suggestion
                    CoflMod
```

**Review by Ekwav — CHANGES_REQUESTED** [review]

File: components/ModDetails/ModDetails.tsx (hunk 10 above)
Comment (matthias-luger) [reply components/ModDetails/ModDetails.tsx:43]: Why? There is the option to full-screen it.
Also: I can't detect if you play the video, just if you somewhere clicked on it.
I would not expect a embeded player to change size when I click on it.

File: components/NavBar/NavBar.tsx [hunk 13]
```diff
@@ -183,6 +184,9 @@ function NavBar(props: Props) {
                         <MenuItem icon={<PetsIcon />}>
                             <Link href={'/kat'}>Kat flips</Link>
                         </MenuItem>
+                        <MenuItem icon={<FileIcon />}>
```
Comment (Ekwav) [inline components/NavBar/NavBar.tsx]: please use a download icon

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Review by Ekwav — APPROVED** [review]

**Pull request merged**